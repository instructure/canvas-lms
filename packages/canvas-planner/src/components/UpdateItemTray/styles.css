.root {
  background: var(--background);
  width: 19rem;
  max-width: 100%;
  box-sizing: border-box;
}
