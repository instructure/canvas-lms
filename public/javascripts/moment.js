define(['bower/moment/min/moment-with-locales'], function(moment) {
  // Include locale that is not a part of momentjs proper.
  var mi_nz = moment.defineLocale('mi-nz', {
    months: 'Kohi-tāte_Hui-tanguru_Poutū-te-rangi_Paenga-whāwhā_Haratua_Pipiri_Hōngoingoi_Here-turi-kōkā_Mahuru_Whiringa-ā-nuku_Whiringa-ā-rangi_Hakihea'.split('_'),
    monthsShort: 'Kohi_Hui_Pou_Pae_Hara_Pipi_Hōngoi_Here_Mahu_Whi-nu_Whi-ra_Haki'.split('_'),
    weekdays: 'Rātapu_Mane_Tūrei_Wenerei_Tāite_Paraire_Hātarei'.split('_'),
    weekdaysShort: 'Ta_Ma_Tū_We_Tāi_Pa_Hā'.split('_'),
    weekdaysMin: 'Ta_Ma_Tū_We_Tāi_Pa_Hā'.split('_'),
    longDateFormat: {
      LT: 'HH:mm',
      LTS: 'HH:mm:ss',
      L: 'DD/MM/YYYY',
      LL: 'D MMMM YYYY',
      LLL: 'D MMMM YYYY [i] HH:mm',
      LLLL: 'dddd, D MMMM YYYY [i] HH:mm'
    },
    calendar: {
      sameDay: '[i teie mahana, i] LT',
      nextDay: '[apopo i] LT',
      nextWeek: 'dddd [i] LT',
      lastDay: '[inanahi i] LT',
      lastWeek: 'dddd [whakamutunga i] LT',
      sameElse: 'L'
    },
    relativeTime: {
      future: 'i roto i %s',
      past: '%s i mua',
      s: 'te hēkona ruarua',
      m: 'he meneti',
      mm: '%d meneti',
      h: 'te haora',
      hh: '%d haora',
      d: 'he ra',
      dd: '%d ra',
      M: 'he marama',
      MM: '%d marama',
      y: 'he tau',
      yy: '%d tau'
    },
    ordinalParse: /\d{1,2}º/,
    ordinal: '%dº'
  });

  return moment;
});
