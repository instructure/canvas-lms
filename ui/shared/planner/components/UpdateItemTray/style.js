/*
 * Copyright (C) 2023 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import {getThemeVars} from '../../getThemeVars'

export default function buildStyle() {
  /*
   * If the theme variables to be used when generating the styles below
   * are dependent on the actual theme in use, you can also pull out the
   * `key` property from the return from `getThemeVars()` and do a bit of
   * if or switch statement logic to get the result you want.
   */
  const {variables} = getThemeVars()

  const theme = {
    background: variables.colors.contrasts.white1010,
  }

  const classNames = {
    root: 'UpdateItemTray-styles__root',
  }

  const css = `
  .${classNames.root} {
    background: ${theme.background};
    width: 19rem;
    max-width: 100%;
    box-sizing: border-box;
  }
  `
  return {css, classNames}
}
