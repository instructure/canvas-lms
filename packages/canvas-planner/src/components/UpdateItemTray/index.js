/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React, { Component } from 'react';
import _ from 'lodash';
import themeable from '@instructure/ui-themeable/lib';
import Container from '@instructure/ui-core/lib/components/Container';
import FormFieldGroup from '@instructure/ui-core/lib/components/FormFieldGroup';
import ScreenReaderContent from '@instructure/ui-core/lib/components/ScreenReaderContent';
import Button from '@instructure/ui-core/lib/components/Button';
import formatMessage from '../../format-message';
import PropTypes from 'prop-types';
import TextInput from '@instructure/ui-core/lib/components/TextInput';
import Select from '@instructure/ui-core/lib/components/Select';
import TextArea from '@instructure/ui-core/lib/components/TextArea';
import DateInput from '@instructure/ui-core/lib/components/DateInput';
import moment from 'moment-timezone';

import { courseShape } from '../plannerPropTypes';
import styles from './styles.css';
import theme from './theme.js';

export class UpdateItemTray extends Component {
  static propTypes = {
    courses: PropTypes.arrayOf(PropTypes.shape(courseShape)).isRequired,
    noteItem: PropTypes.object,
    onSavePlannerItem: PropTypes.func.isRequired,
    onDeletePlannerItem: PropTypes.func.isRequired,
    locale: PropTypes.string.isRequired,
    timeZone: PropTypes.string.isRequired,
  };

  constructor (props) {
    super(props);
    const updates = this.getNoteUpdates(props);
    if (!updates.date) {
      updates.date = props.noteItem && props.noteItem.date ? props.noteItem.date : moment.tz(props.timeZone).toISOString();
    }
    this.state = {
      updates,
      titleMessages: [],
      dateMessages: [],
    };
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.noteItem) {
      const updates = this.getNoteUpdates(nextProps);
      this.setState({updates}, this.updateMessages);
    }
  }

  getNoteUpdates (props) {
    const updates = _.cloneDeep(props.noteItem) || {};
    if (updates.context) {
      updates.courseId = updates.context.id;
      delete updates.context;
    }
    return updates;
  }

  updateMessages = () => {
    if (!this.state.updates.date) {
      this.setState({dateMessages: [{type: 'error', text: formatMessage('Date is required')}]});
    } else {
      this.setState({dateMessages: []});
    }
  }

  handleSave = () => {
    const updates = Object.assign({}, this.state.updates);
    if (updates.courseId) {
      updates.context = { id: updates.courseId };
    } else {
      updates.context = { id: null };
    }
    delete updates.courseId;
    this.props.onSavePlannerItem(updates);
  }

  handleChange = (field, value) => {
    this.setState({
      updates: {
        ...this.state.updates,
        [field]: value
      }
    }, this.updateMessages);
  }

  handleCourseIdChange = (e) => {
    let value = e.target.value;
    if (value === 'none') value = undefined;
    this.handleChange('courseId', value);
  }

  handleTitleChange = (e) => {
    const value = e.target.value;
    if (value === '') {
      this.setState({
        titleMessages: [{type: 'error', text: formatMessage('title is required')}]
      });
    } else {
        this.setState({titleMessages: []});
    }
    this.handleChange('title', value);
  }

  handleDateChange = (e, date) => {
    const value = date ? moment(date).toISOString() : '';  // works if date is a moment obj (instui 3) or iso string (instui 4)
      this.handleChange('date', value);
  }

  handleDeleteClick = () => {
    // eslint-disable-next-line no-restricted-globals
    if (confirm(formatMessage('Are you sure you want to delete this planner item?'))) {
      this.props.onDeletePlannerItem(this.props.noteItem);
    }
  }

  findCurrentValue (field) {
    return this.state.updates[field] || '';
  }

  isValid () {
    if (this.state.updates.title && this.state.updates.date) {
      return this.state.updates.title.replace(/\s/g, '').length > 0;
    }
    return false;
  }

  renderDeleteButton () {
    if (this.props.noteItem == null) return;
    return <Button
      variant="light"
      margin="0 x-small 0 0"
      onClick={this.handleDeleteClick}>
      {formatMessage("Delete")}
    </Button>;
  }

  renderSaveButton () {
    return <Button
      variant="primary"
      margin="0 0 0 x-small"
      disabled={!this.isValid()}
      onClick={this.handleSave}>
        {formatMessage("Save")}
    </Button>;
  }

  renderTitleInput () {
    const value = this.findCurrentValue('title');
    return (
      <TextInput
        label={formatMessage("Title")}
        value={value}
        messages={this.state.titleMessages}
        onChange={this.handleTitleChange}
      />
    );
  }

  renderDateInput () {
    return (
      <DateInput
        required={true}
        messages={this.state.dateMessages}
        label={formatMessage("Date")}
        nextLabel={formatMessage("Next Month")}
        previousLabel={formatMessage("Previous Month")}
        locale={this.props.locale}
        timezone={this.props.timeZone}
        placement="start"
        dateValue={this.state.updates.date}
        onDateChange={this.handleDateChange}
      />
    );
  }

  renderCourseSelectOptions () {
    if (!this.props.courses) return [];
    return this.props.courses.map(course => {
      return <option key={course.id} value={course.id}>{course.longName}</option>;
    });
  }

  renderCourseSelect () {
    let courseId = this.findCurrentValue('courseId');
    if (courseId == null) courseId = 'none';
    return (
      <Select
        label={formatMessage("Course")}
        value={courseId}
        onChange={this.handleCourseIdChange}
      >
        <option value="none">{formatMessage("Optional: Add Course")}</option>
        {this.renderCourseSelectOptions()}
      </Select>
    );
  }

  renderDetailsInput () {
    const value = this.findCurrentValue('details');
    return (
      <TextArea
        label={formatMessage("Details")}
        height="10rem"
        autoGrow={false}
        value={value}
        onChange={(e) => this.handleChange('details', e.target.value)}
      />
    );
  }

  renderTrayHeader () {
    if (this.props.noteItem) {
      return (
        <h2>{formatMessage('Edit {title}', { title: this.props.noteItem.title })}</h2>
      );
    } else {
      return (
        <h2>{formatMessage("Add To Do")}</h2>
      );
    }
  }

  render () {
    return (
      <div className={styles.root}>
        <Container
          as="div"
          padding="large medium medium"
        >
          <FormFieldGroup
            rowSpacing="small"
            description={
              <ScreenReaderContent>
                {this.renderTrayHeader()}
              </ScreenReaderContent>
            }
          >
            {this.renderTitleInput()}
            {this.renderDateInput()}
            {this.renderCourseSelect()}
            {this.renderDetailsInput()}
          </FormFieldGroup>
          <Container as="div" margin="small 0 0" textAlign="end">
            {this.renderDeleteButton()}
            {this.renderSaveButton()}
          </Container>
        </Container>
      </div>
    );
  }
}

export default themeable(theme, styles)(UpdateItemTray);
