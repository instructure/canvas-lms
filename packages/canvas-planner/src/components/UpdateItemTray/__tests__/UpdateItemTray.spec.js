/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import moment from 'moment-timezone';
import React from 'react';
import { shallow, mount } from 'enzyme';
import UpdateItemTray from '../index';

const defaultProps = {
  onSavePlannerItem: () => {},
  locale: 'en',
  timeZone: 'Asia/Tokyo',
  onDeletePlannerItem: () => {},
  courses: [],
};

it('renders the item to update if provided', () => {
  const noteItem = {
    title: 'Planner Item',
    date: '2017-04-25 01:49:00-0700',
    context: {id: '1'},
    details: "You made this item to remind you of something, but you forgot what."
  };
  const wrapper = shallow(
    <UpdateItemTray {...defaultProps}
      noteItem={noteItem}
      courses={[{id: '1', longName: 'a course'}]} />
  );
  expect(wrapper).toMatchSnapshot();
});

it("doesn't re-render unless new item is provided", () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} />)
  const newProps = {...defaultProps, locale: 'fr'}
  wrapper.setProps(newProps)
  expect(wrapper.find('DateInput').props()['messages'].length).toBe(0)
})

it('renders Add To Do header when creating a new to do', () => {
  const wrapper = mount(
    <UpdateItemTray {...defaultProps} />
  );

  expect(wrapper.find('h2').text()).toBe('Add To Do');
});

it('shows title inputs', () => {
  const wrapper = mount(
    <UpdateItemTray {...defaultProps} />
  );
  expect(wrapper.find('TextInput')).toHaveLength(2);
  const input = wrapper.find('TextInput').first();
  input.find('input').simulate('change', {target: {value: 'New Text'}});
  expect(input.props().value).toEqual('New Text');
});

it('shows details inputs', () => {
  const wrapper = mount(
    <UpdateItemTray {...defaultProps} />
  );
  const input = wrapper.find('TextArea');
  input.find('textarea').simulate('change', {target: {value: 'New Details'}});
  expect(input.props().value).toEqual('New Details');
});

it('disables the save button when title is empty', () => {
  const item = { title: '', date: '2017-04-28' };
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={item} />);
  const button = wrapper.find('Button[variant="primary"]');
  expect(button.props().disabled).toBe(true);
});

it('handles courseid being none', () => {
  const item = { title: '', date: '2017-04-28' };
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={item} />);
  wrapper.instance().handleCourseIdChange({target: {value: 'none'}});
  expect(wrapper.instance().state.updates.courseId).toBe(undefined);
});

it('correctly updates id to null when courseid is none', () => {
  const item = { title: '', date: '2017-04-28' };
  const mockCallback = jest.fn();
  const wrapper = shallow(<UpdateItemTray {...defaultProps} onSavePlannerItem={mockCallback} noteItem={item} />);
  wrapper.instance().handleCourseIdChange({target: {value: 'none'}});
  wrapper.instance().handleSave();
  expect(mockCallback).toHaveBeenCalledWith({
    title: item.title,
    date: item.date,
    context: {
      id: null
    }
  });
});

it('sets default date when no date is provided', () => {
  const item = { title: 'an item', date: '' };
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={item} />);
  const datePicker = wrapper.find('DateInput');
  expect(!datePicker.props().dateValue.length).toBe(false);
});

it('enables the save button when title and date are present', () => {
  const item = { title: 'an item', date: '2017-04-28' };
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={item} />);
  const button = wrapper.find('Button[variant="primary"]');
  expect(button.props().disabled).toBe(false);
});

it('does not set an initial error message on title', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} />);
  const titleInput = wrapper.find('TextInput').first();
  expect(titleInput.props().messages).toEqual([]);
});

it('sets error message on title field when title is set to blank', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={{title: 'an item'}} />);
  wrapper.instance().handleTitleChange({target: {value: ''}});
  const titleInput = wrapper.find('TextInput').first();
  const messages = titleInput.props().messages;
  expect(messages).toHaveLength(1);
  expect(messages[0].type).toBe('error');
});

it('clears the error message when a title is typed in', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={{title: 'an item'}} />);
  wrapper.instance().handleTitleChange({target: {value: ''}});
  wrapper.instance().handleTitleChange({target: {value: 't'}});
  const titleInput = wrapper.find('TextInput').first();
  expect(titleInput.props().messages).toEqual([]);
});

// The Date picker does not support error handling yet we are working with instui to get it working
xit('does not set an initial error message on date', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} />);
  const dateInput = wrapper.find('TextInput').at(1);
  expect(dateInput.props().messages).toEqual([]);
});

xit('sets error message on date field when date is set to blank', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={{date: '2017-04-28'}} />);
  wrapper.instance().handleDateChange({target: {value: ''}});
  const dateInput = wrapper.find('TextInput').at(1);
  const messages = dateInput.props().messages;
  expect(messages).toHaveLength(1);
  expect(messages[0].type).toBe('error');
});

xit('clears the error message when a date is typed in', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={{date: '2017-04-28'}} />);
  wrapper.instance().handleTitleChange({target: {value: ''}});
  wrapper.instance().handleTitleChange({target: {value: '2'}});
  const dateInput = wrapper.find('TextInput').at(1);
  expect(dateInput.props().messages).toEqual([]);
});

it('respects the provided timezone', () => {
  const item = { title: '', date: '2017-04-25 12:00:00-0300' };
  const wrapper = mount(<UpdateItemTray {...defaultProps} noteItem={item} />);
  const d = wrapper.find('DateInput').find('TextInput').props().value;
  expect(d).toEqual('April 26, 2017');  // timezone shift from -3 to +9 pushes it to the next day
});

it('changes state when new date is typed in', () => {
  const noteItem = {
    title: 'Planner Item',
    date: '2017-04-25',
  };
  const mockCallback = jest.fn();
  const wrapper = mount(<UpdateItemTray {...defaultProps} onSavePlannerItem={mockCallback} noteItem={noteItem} />);
  const newDate = moment('2017-10-16');
  wrapper.instance().handleDateChange({}, newDate.toISOString());
  wrapper.instance().handleSave();
  expect(mockCallback).toHaveBeenCalledWith({
    title: noteItem.title,
    date: newDate.toISOString(),
    context: {
      id: null
    }
  });
});

it('updates state when new note is passed in', () => {
  const noteItem1 = {
    title: 'Planner Item 1',
    date: '2017-04-25',
    context: {id: '1'},
    details: "You made this item to remind you of something, but you forgot what."
  };
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={noteItem1} courses={[
    {id: '1', longName: 'first course'},
    {id: '2', longName: 'second course'},
  ]}/>);
  expect(wrapper).toMatchSnapshot();

  const noteItem2 = {
    title: 'Planner Item 2',
    date: '2017-12-25',
    context: {id: '2'},
    details: "This is another reminder"
  };
  wrapper.setProps({noteItem: noteItem2});
  expect(wrapper).toMatchSnapshot();
});

//------------------------------------------------------------------------

it('does not render the delete button if an item is not specified', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} />);
  const deleteButton = wrapper.find('Button[variant="light"]');
  expect(deleteButton).toHaveLength(0);
});

it('does render the delete button if an item is specified', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} noteItem={{title: 'some note'}} />);
  const deleteButton = wrapper.find('Button[variant="light"]');
  expect(deleteButton).toHaveLength(1);
});

it('renders just an optional option when no courses', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} />);
  expect(wrapper.find('option')).toHaveLength(1);
});

it('renders course options plus an optional option when provided with courses', () => {
  const wrapper = shallow(<UpdateItemTray {...defaultProps} courses={[
    {id: '1', longName: 'first course'},
    {id: '2', longName: 'second course'},
  ]} />);
  expect(wrapper.find('option')).toHaveLength(3);
});

it('invokes save callback with updated data', () => {
  const saveMock = jest.fn();
  const wrapper = shallow(<UpdateItemTray {...defaultProps}
    noteItem={{
      title: 'title', date: '2017-04-27', courseId: '42', details: 'details',
    }}
    courses={[{id: '42', longName: 'first'}, {id: '43', longName: 'second'}]}
    onSavePlannerItem={saveMock}
  />);
  wrapper.instance().handleTitleChange({target: {value: 'new title'}});
  wrapper.instance().handleDateChange({}, '2017-05-01');
  wrapper.instance().handleCourseIdChange({target: {value: '43'}});
  wrapper.instance().handleChange('details', 'new details');
  wrapper.instance().handleSave();
  expect(saveMock).toHaveBeenCalledWith({
    title: 'new title', date: moment('2017-05-01').toISOString(), context: {id: '43'}, details: 'new details',
  });
});

it('invokes the delete callback', () => {
  const mockDelete = jest.fn();
  const wrapper = shallow(<UpdateItemTray {...defaultProps}
    noteItem={{title: 'a title'}}
    onDeletePlannerItem={mockDelete} />);
  const confirmSpy = jest.spyOn(window, 'confirm').mockReturnValue(true);
  wrapper.instance().handleDeleteClick();
  expect(confirmSpy).toHaveBeenCalled();
  expect(mockDelete).toHaveBeenCalledWith({title: 'a title'});
  confirmSpy.mockRestore();
});
