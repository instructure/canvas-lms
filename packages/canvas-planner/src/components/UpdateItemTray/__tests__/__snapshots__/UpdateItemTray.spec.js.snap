// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the item to update if provided 1`] = `
<div>
  <Container
    as="div"
    display={null}
    padding="large medium medium"
  >
    <FormFieldGroup
      as="fieldset"
      colSpacing="small"
      description={
        <ScreenReaderContent
          as="span"
        >
          <h2>
            Edit Planner Item
          </h2>
        </ScreenReaderContent>
      }
      disabled={false}
      rowSpacing="small"
      vAlign="middle"
    >
      <TextInput
        disabled={false}
        inline={false}
        inputRef={[Function]}
        label="Title"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        size="medium"
        textAlign="start"
        type="text"
        value="Planner Item"
      />
      <DateInput
        datePickerRef={[Function]}
        dateValue="2017-04-25 01:49:00-0700"
        disabled={false}
        format="LL"
        inline={false}
        inputRef={[Function]}
        invalidDateMessage={[Function]}
        label="Date"
        layout="stacked"
        locale="en"
        messages={Array []}
        nextLabel="Next Month"
        onDateChange={[Function]}
        placement="start"
        previousLabel="Previous Month"
        required={true}
        size="medium"
        timezone="Asia/Tokyo"
        validationFeedback={true}
      />
      <Select
        disabled={false}
        inline={false}
        label="Course"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        selectRef={[Function]}
        size="medium"
        type="text"
        value="1"
      >
        <option
          value="none"
        >
          Optional: Add Course
        </option>
        <option
          value="1"
        >
          a course
        </option>
      </Select>
      <TextArea
        autoGrow={false}
        disabled={false}
        height="10rem"
        inline={false}
        label="Details"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        resize="none"
        size="medium"
        textareaRef={[Function]}
        value="You made this item to remind you of something, but you forgot what."
      />
    </FormFieldGroup>
    <Container
      as="div"
      display={null}
      margin="small 0 0"
      textAlign="end"
    >
      <Button
        as="button"
        buttonRef={[Function]}
        fluidWidth={false}
        margin="0 x-small 0 0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="light"
      >
        Delete
      </Button>
      <Button
        as="button"
        buttonRef={[Function]}
        disabled={false}
        fluidWidth={false}
        margin="0 0 0 x-small"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="primary"
      >
        Save
      </Button>
    </Container>
  </Container>
</div>
`;

exports[`updates state when new note is passed in 1`] = `
<div>
  <Container
    as="div"
    display={null}
    padding="large medium medium"
  >
    <FormFieldGroup
      as="fieldset"
      colSpacing="small"
      description={
        <ScreenReaderContent
          as="span"
        >
          <h2>
            Edit Planner Item 1
          </h2>
        </ScreenReaderContent>
      }
      disabled={false}
      rowSpacing="small"
      vAlign="middle"
    >
      <TextInput
        disabled={false}
        inline={false}
        inputRef={[Function]}
        label="Title"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        size="medium"
        textAlign="start"
        type="text"
        value="Planner Item 1"
      />
      <DateInput
        datePickerRef={[Function]}
        dateValue="2017-04-25"
        disabled={false}
        format="LL"
        inline={false}
        inputRef={[Function]}
        invalidDateMessage={[Function]}
        label="Date"
        layout="stacked"
        locale="en"
        messages={Array []}
        nextLabel="Next Month"
        onDateChange={[Function]}
        placement="start"
        previousLabel="Previous Month"
        required={true}
        size="medium"
        timezone="Asia/Tokyo"
        validationFeedback={true}
      />
      <Select
        disabled={false}
        inline={false}
        label="Course"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        selectRef={[Function]}
        size="medium"
        type="text"
        value="1"
      >
        <option
          value="none"
        >
          Optional: Add Course
        </option>
        <option
          value="1"
        >
          first course
        </option>
        <option
          value="2"
        >
          second course
        </option>
      </Select>
      <TextArea
        autoGrow={false}
        disabled={false}
        height="10rem"
        inline={false}
        label="Details"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        resize="none"
        size="medium"
        textareaRef={[Function]}
        value="You made this item to remind you of something, but you forgot what."
      />
    </FormFieldGroup>
    <Container
      as="div"
      display={null}
      margin="small 0 0"
      textAlign="end"
    >
      <Button
        as="button"
        buttonRef={[Function]}
        fluidWidth={false}
        margin="0 x-small 0 0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="light"
      >
        Delete
      </Button>
      <Button
        as="button"
        buttonRef={[Function]}
        disabled={false}
        fluidWidth={false}
        margin="0 0 0 x-small"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="primary"
      >
        Save
      </Button>
    </Container>
  </Container>
</div>
`;

exports[`updates state when new note is passed in 2`] = `
<div>
  <Container
    as="div"
    display={null}
    padding="large medium medium"
  >
    <FormFieldGroup
      as="fieldset"
      colSpacing="small"
      description={
        <ScreenReaderContent
          as="span"
        >
          <h2>
            Edit Planner Item 2
          </h2>
        </ScreenReaderContent>
      }
      disabled={false}
      rowSpacing="small"
      vAlign="middle"
    >
      <TextInput
        disabled={false}
        inline={false}
        inputRef={[Function]}
        label="Title"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        size="medium"
        textAlign="start"
        type="text"
        value="Planner Item 2"
      />
      <DateInput
        datePickerRef={[Function]}
        dateValue="2017-12-25"
        disabled={false}
        format="LL"
        inline={false}
        inputRef={[Function]}
        invalidDateMessage={[Function]}
        label="Date"
        layout="stacked"
        locale="en"
        messages={Array []}
        nextLabel="Next Month"
        onDateChange={[Function]}
        placement="start"
        previousLabel="Previous Month"
        required={true}
        size="medium"
        timezone="Asia/Tokyo"
        validationFeedback={true}
      />
      <Select
        disabled={false}
        inline={false}
        label="Course"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        selectRef={[Function]}
        size="medium"
        type="text"
        value="2"
      >
        <option
          value="none"
        >
          Optional: Add Course
        </option>
        <option
          value="1"
        >
          first course
        </option>
        <option
          value="2"
        >
          second course
        </option>
      </Select>
      <TextArea
        autoGrow={false}
        disabled={false}
        height="10rem"
        inline={false}
        label="Details"
        layout="stacked"
        messages={Array []}
        onChange={[Function]}
        resize="none"
        size="medium"
        textareaRef={[Function]}
        value="This is another reminder"
      />
    </FormFieldGroup>
    <Container
      as="div"
      display={null}
      margin="small 0 0"
      textAlign="end"
    >
      <Button
        as="button"
        buttonRef={[Function]}
        fluidWidth={false}
        margin="0 x-small 0 0"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="light"
      >
        Delete
      </Button>
      <Button
        as="button"
        buttonRef={[Function]}
        disabled={false}
        fluidWidth={false}
        margin="0 0 0 x-small"
        onClick={[Function]}
        size="medium"
        type="button"
        variant="primary"
      >
        Save
      </Button>
    </Container>
  </Container>
</div>
`;
